WordPress/Classes/Services/AccountService+MergeDuplicates.swift

```diff
@@ -3,22 +3,20 @@ import Foundation
 extension AccountService {
 
     func mergeDuplicatesIfNecessary() {
-        guard let count = try? WPAccount.lookupNumberOfAccounts(in: managedObjectContext), count > 1 else {
-            return
-        }
-
-        let accounts = allAccounts()
-        let accountGroups = Dictionary(grouping: accounts) { $0.userID }
-        for group in accountGroups.values where group.count > 1 {
-            mergeDuplicateAccounts(accounts: group)
-        }
-
-        if managedObjectContext.hasChanges {
-            ContextManager.sharedInstance().save(managedObjectContext)
+        coreDataStack.performAndSave { context in
+            guard let count = try? WPAccount.lookupNumberOfAccounts(in: context), count > 1 else {
```

Comment:
I guess some day we'll probably want a `throws`-ing variant of `performAndSave` 😅

Reply:
A while ago, I tried to get the current one to `rethrow`, but I don't think Swift compiler can handle an "escaped" rethrow block. We probably will have to have a `performAndSaveThrowing` variant, instead of sharing the reusing the same function "performAndSave" name.